x/staking/handler.go

```diff
@@ -0,0 +1,54 @@
+package staking
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/desmos-labs/juno/parse/client"
+	"github.com/desmos-labs/juno/parse/worker"
+	"github.com/desmos-labs/juno/types"
+	"github.com/forbole/bdjuno/database"
+	"github.com/rs/zerolog/log"
+
+	"fmt"
+)
+
+func MsgHandler(tx types.Tx, index int, msg sdk.Msg, w worker.Worker) error {
+	log.Info().Str("tx_hash", tx.TxHash).Int("msg_index", index).Str("msg_type", msg.Type()).Msg("found message")
+
+	if len(tx.Logs) == 0 {
+		log.Info().Msg(fmt.Sprintf("Skipping message at index %d of tx hash %s as it was not successull",
+			index, tx.TxHash))
+		return nil
+	}
+	db, ok := w.Db.(database.BigDipperDb)
+	if !ok {
+		return fmt.Errorf("invalid database")
+	}
+	switch stakingMsg := msg.(type) {
+	case staking.MsgEditValidator:
+		// TODO: Handle message here
+		//store commission rate
+		StoreModifiedCommission(stakingMsg, w.ClientProxy, db)
+	}
+
+	return nil
+}
+
+func StoreModifiedCommission(msg staking.MsgEditValidator, cp client.ClientProxy, db database.BigDipperDb) error {
+	//should I take from REST or store the message?
+	//store the message
+	address := msg.ValidatorAddress
+	if found, _ := db.HasValidator(address.String()); !found {
+		return nil
+	}
+
+	var validator staking.Validator
+	endpoint := fmt.Sprintf("/staking/validators/$s", address.String())
+	height, ok := cp.QueryLCDWithHeight(endpoint, &validator)
+	if ok != nil {
+		return ok
+	}
```

Comment:
When should I take the message from LCD and from msg?
Kwan said I should take from LCD when getting the latest state of the chain.

Reply:
> When should I take the message from LCD and from msg?

Since the LCD has some overhead associated to it, if possible always use the data present inside the message. 

Messages inside transactions are always parsed, even if you do not use them, cause they are also sent to other 
`MsgHandler` methods, so registering a new `MsgHandler` and using the data inside the messages has ideally no overhead at all. 

On the other hand, querying an LCD means that a network request must be done and the response must be de-serialized twice: one to get the response and another one to get its content.

---

> Kwan said I should take from LCD when getting the latest state of the chain.

LCD requests should be done only when querying the data that cannot be queried in any other way (this is the case of data that is not changed thought messages such as the staking pool or the staking rewards), and not when the same data can be take from blocks/transactions/messages/